### PR TITLE
Add GitHub action to generate code report #968

### PR DESCRIPTION
**Pull request by GracianoR**
Signed-off-by: GracianoR <rodrigo.graciano@bnymellon.com>

### CODE REVIEW

**Comment by mohrezaei** [comment]
I know there were some questions in the Issue, but now that there is a PR we can move the discussion here.

I would be in favor of doing this on every PR and integrating with [Coveralls](https://coveralls.io).

Output will look like [this](https://github.com/apache/commons-math/pull/156).

You can also look at that [project](https://github.com/apache/commons-math) to see how it's setup in pom.xml using this [plugin](https://github.com/trautonen/coveralls-maven-plugin).

**Comment by GracianoR** [comment]
Thank you @mohrezaei. I also tried before using Codecov but I believe both of them would require creating an account at Coveralls/Codecov. This is an option as well and has the advantage of the badge ready to be used as well.

**Comment by mohrezaei** [comment]
Creating an account shouldn't be a problem (I can do it for the project). I prefer coveralls based on the legalize.

**Comment by mohrezaei** [comment]
Looks like coveralls/maven plugin [doesn't work](https://github.com/trautonen/coveralls-maven-plugin/issues/136) with github actions, so I guess we can forget all that ☹️

**Comment by donraab** [comment]
I think we should just get the report generating to start so we can start identifying any coverage paths developers can work on. The build for that can stay scheduled to start. Eventually it would be great to have something like you linked to above with Coveralls and to build code coverage on every PR. I suspect Coveralls will eventually have support for GitHub Actions.

Thoughts?

File: .github/workflows/coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
```
Comment (prathasirisha) [inline .github/workflows/coverage.yaml:14]: Can remove extra blank lines? Line 12, Line 14.

File: .github/workflows/coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 1.8
```
Comment (prathasirisha) [inline .github/workflows/coverage.yaml:17]: In other GitHub action yaml files, we have java-version configured as a variable that gets used in line number 17 and 20. 
Though what you have works, can you make it consistent with other yaml files? This is just a  personal preference.

File: .github/workflows/coverage.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Coverage Report
+
+on:
+  schedule:
+    - cron: '0 0 * * 0'
```
[the remark this answers is not in the document]
Comment (GracianoR) [reply .github/workflows/coverage.yaml:8]: I should have mentioned that

File: .github/workflows/coverage.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set JDK ${{ matrix.java-version }}
```
Comment (prathasirisha) [inline .github/workflows/coverage.yaml:15]: @GracianoR I could be wrong, but I expected a variable definition like this for this to work. 
`matrix:`    
        `os: [ubuntu-latest]`  
        `java-version: [8]`   
See example [here](https://github.com/eclipse/eclipse-collections/blob/master/.github/workflows/acceptance.yml) - line 12-14

**Comment by GracianoR** [comment]
@prathasirisha, changes were done and ready to be reviewed

**Comment by donraab** [comment]
Thanks for the contribution @GracianoR and thank you for reviewing the PR @prathasirisha !

**Pull request merged**